pkg/clients/secretsmanager/secret.go

```diff
@@ -0,0 +1,267 @@
+func NewSecretsmanagerClient(cfg aws.Config) Client {
+	return secretsmanager.New(cfg)
+}
+
+// IsErrorNotFound helper function to test for ErrCodeResourceNotFoundException error.
+func IsErrorNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), secretsmanager.ErrCodeResourceNotFoundException)
+}
+
+// GetSecret fetches the referenced secret for a Secret CRD
+func GetSecret(ctx context.Context, kube client.Client, s *v1alpha1.Secret) (secret []byte, err error) {
+	nn := types.NamespacedName{
+		Name:      s.Spec.ForProvider.SecretBinarySecretRef.Name,
+		Namespace: s.Spec.ForProvider.SecretBinarySecretRef.Namespace,
+	}
+	sc := &corev1.Secret{}
+	if err := kube.Get(ctx, nn, sc); err != nil {
+		return nil, errors.Wrap(err, errGetSecretFailed)
+	}
+	if len(sc.Data) == 0 {
+		return nil, nil
+	}
+	scb, err := json.Marshal(sc.Data)
+	if err != nil {
+		return nil, errors.New(errSecretMarshalFailed)
+	}
+	return scb, nil
+}
+
+// GenerateCreateSecretsmanagerInput from ClusterParameters.
+func GenerateCreateSecretsmanagerInput(name string, p *v1alpha1.SecretParameters, secret []byte) *secretsmanager.CreateSecretInput {
+	c := &secretsmanager.CreateSecretInput{
+		Name:         awsclients.String(name),
+		SecretBinary: secret,
+	}
+
+	if p.Description != nil {
+		c.Description = p.Description
+	}
+
+	if p.KmsKeyID != nil {
+		c.KmsKeyId = p.KmsKeyID
+	}
+
+	if len(p.Tags) != 0 {
+		c.Tags = make([]secretsmanager.Tag, len(p.Tags))
+		for i, val := range p.Tags {
+			c.Tags[i] = secretsmanager.Tag{
+				Key:   aws.String(val.Key),
+				Value: aws.String(val.Value),
+			}
+		}
+	}
+
+	return c
+}
+
+// GenerateUpdateSecretInput from SecretParameters
+func GenerateUpdateSecretInput(name string, p v1alpha1.SecretParameters, secret []byte) *secretsmanager.UpdateSecretInput {
+	u := &secretsmanager.UpdateSecretInput{
+		SecretBinary: secret,
+		SecretId:     awsclients.String(name),
+	}
+
+	if p.Description != nil {
+		u.Description = p.Description
+	}
+
+	if p.KmsKeyID != nil {
+		u.KmsKeyId = p.KmsKeyID
+	}
+
+	return u
+}
+
+// GenerateDeleteSecretInput from SecretParameters
+func GenerateDeleteSecretInput(name string, p v1alpha1.SecretParameters) *secretsmanager.DeleteSecretInput {
+	d := &secretsmanager.DeleteSecretInput{
+		SecretId: awsclients.String(name),
+	}
+
+	switch r := GetRecoveryWindowInDays(p.RecoveryWindowInDays); r {
+	case 0:
+		d.ForceDeleteWithoutRecovery = boolToPtr(true)
+	default:
+		d.RecoveryWindowInDays = IntToPtr(int64(r))
+	}
+
+	return d
+}
+
+// UpdateObservation updates status.AtProvider
+func UpdateObservation(o *v1alpha1.SecretObservation, svr *secretsmanager.GetSecretValueResponse, do *secretsmanager.DescribeSecretOutput) {
+	if svr != nil {
+		svo := svr.GetSecretValueOutput
+		if svo != nil {
+			arn := svo.ARN
+			if svo.ARN == nil {
+				arn = do.ARN
+			}
+			o.ARN = awsclients.StringValue(arn)
+			o.VersionID = awsclients.StringValue(svo.VersionId)
```

Comment:
This seems to be set at every reconcile, hence triggers a new reconcile, so it reconciles repeatedly. Can we remove this field if not used in a functional logic in the code?